awx/ui/src/screens/Inventory/shared/Inventory.helptext.js

```diff
@@ -0,0 +1,192 @@
+/* eslint-disable react/destructuring-assignment */
+import React from 'react';
+import { t, Trans } from '@lingui/macro';
+import { Link } from 'react-router-dom';
+
+const docUrls = {
+  ec2: 'https://docs.ansible.com/ansible/latest/collections/amazon/aws/aws_ec2_inventory.html',
+  azure_rm:
+    'https://docs.ansible.com/ansible/latest/collections/azure/azcollection/azure_rm_inventory.html',
+  controller:
+    'https://docs.ansible.com/ansible/latest/collections/awx/awx/tower_inventory.html',
+  gce: 'https://docs.ansible.com/ansible/latest/collections/google/cloud/gcp_compute_inventory.html',
+  insights:
+    'https://docs.ansible.com/ansible/latest/collections/redhatinsights/insights/insights_inventory.html',
+  openstack:
+    'https://docs.ansible.com/ansible/latest/collections/openstack/cloud/openstack_inventory.html',
+  satellite6:
+    'https://docs.ansible.com/ansible/latest/collections/theforeman/foreman/foreman_inventory.html',
+  rhv: 'https://docs.ansible.com/ansible/latest/collections/ovirt/ovirt/ovirt_inventory.html',
+  vmware:
+    'https://docs.ansible.com/ansible/latest/collections/community/vmware/vmware_vm_inventory_inventory.html',
+};
+
+function getInventoryHelpTextStrings(source, docsBaseUrl, projectField = {}) {
```

Comment:
I would suggest making this function an object literal with functions that take the necessary arguments to render the string values that we need instead. The different ways to call this function make it obscure for me as to its intended usage/purpose:

```
Method 1:
const helpText = getInventoryHelpTextStrings();

Method 2:
const helpText = getInventoryHelpTextStrings(
    'azure_rm',
    `${getDocsBaseUrl(
      config
    )}/html/userguide/inventories.html#inventory-plugins`
  );

Method 3:
const helpText = InventoryHelpText(null, projectField);
```

We could make `sourceVars, updateOnLaunch, ...` functions that takes args and call it in-line in the render method of the component itself, e.g.: 
```
<SourceVarsField popoverContent={helpText.sourceVars('azure_rm', docsLink)} />
```

Reply:
yeah, I like this approach